src/models.rs

```diff
@@ -160,6 +160,22 @@ fn default_root_path_value() -> String {
     "/".to_string()
 }
 
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Hash)]
+pub enum Protocol {
+    HTTP,
+}
+
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Hash)]
+pub struct Port {
```

Comment:
Maybe create a constructor on `Port`, it will allow you to omit `publicly_accessible` field since it can be computed from `public_port.is_some()`.

Reply:
This is the struct/payload received by NATS/core